hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/CalciteSqlOptimizer.java

```diff
@@ -233,6 +233,7 @@ public CalciteSqlOptimizer(NodeEngine nodeEngine, QueryResultRegistry resultRegi
         this.planExecutor = new PlanExecutor(
                 tableResolverImpl,
                 dataLinksResolver,
+                nodeEngine,
```

Comment:
we do not need to pass both `nodeEngine` and `nodeEngine.getHazelcastInstance()`

Reply:
We do because of `PlanExecutorTest` that's mocking them.

Reply:
you can setup the mocks like that: `when(nodeEngine.getHazelcastInstance()).thenReturn(hazelcastInstance)`

Reply:
Done. Had to remove `@InjectMocks` annotation, because I needed more control over the order of mock creation and `willReturn` clauses.